### PR TITLE
Services/api

### PR DESCRIPTION
**Pull request by OlegAndrus12**
image

Those changes introduce the pubsub layer and an example of calling `api/get-all-users/` url

### CODE REVIEW

File: services/api_service/api_service/pubsub.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            except:
+                user_pull.cancel()
+                os._exit(1)
+
+    def publish(self, topic, message, message_id):
+        topic_path = self.__publisher.topic_path(self.__project_id, topic)
+        res = self.__publisher.publish(
+            topic_path, message.encode("utf-8"), message_id=message_id
+        )
+        print(f"published message: {res.result()}")
```
Comment (ZhavoronkovN) [inline services/api_service/api_service/pubsub.py:33]: new line

File: services/api_service/api_service/result_watcher.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        threading.Thread(target = manager.setup_subscriber, args=[subscription, lambda m : self._message_handler(m)]).start()
+
+    def _message_handler(self, message):
+        message.ack()
+        print(f"Received message : {message}")
+        while len(self._results) > self.MAX_ITEMS:
+            self._results.popitem(False)
+        id = self._get_id(message)
+        self._results[id] = json.loads(message.data.decode("utf-8"))
+        self._mutexes.pop(id).release()
```
Comment (ZhavoronkovN) [inline services/api_service/api_service/result_watcher.py:21]: I though we've agreed that we should add check here

File: services/api_service/api_service/views.py [hunk 3]
```diff
@@ -1,10 +1,35 @@
...
+from api_service.result_watcher import ResultWatcher
 
 
+SUBSCRIPTIONS_PATH = "projects/lohfinder-app/subscriptions/"
+PROJECT_ID = "lohfinder-app"
+pubsub_manager = PubSubManager(SUBSCRIPTIONS_PATH, PROJECT_ID)
+users_result_watcher = ResultWatcher(pubsub_manager, "user_service_result-sub", lambda m : m.attributes["message_id"])
+
+def get_id():
+    return str(uuid.uuid4().int & (1<<64)-1)
```
Comment (ZhavoronkovN) [inline services/api_service/api_service/views.py:18]: we need to set seed depending on time here, otherwise numbers will be the same with every execution

Comment (OlegAndrus12) [reply services/api_service/api_service/views.py:18]: uuid.uuid4 generates new numbers each time. Checked it on two different terminals

**Review by ZhavoronkovN — APPROVED** [review]

**Pull request merged**